scripts/apstart.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/sh
+
+LOGFILE="/mnt/mmc/softap.log"
+HOSTAPD_CONF="/opt/usr/nx-ks/ap/hostapd_uap.conf"
+UAP_IF="uap0"
+IP_ADDR="192.168.4.1"
+SUBNET="255.255.255.0"
+UDHCPD_CONF="/opt/usr/nx-ks/ap/udhcp_uap.conf"
+UDHCPD_PID="/var/run/udhcpd.pid"
+LEASES_FILE="/opt/usr/nx-ks/ap/udhcpd.leases"
+
+sleep 5
+
+log() {
+    echo "$(date '+%Y-%m-%d %H:%M:%S') - $1" | tee -a "$LOGFILE"
+}
+
+# Set regulatory domain to Germany
+log "Setting regulatory domain to DE..."
+iw reg set DE >> "$LOGFILE" 2>&1 && log "Regulatory domain set to DE." || log "Failed to set regulatory domain to DE."
+
+# Retrieve and set MAC address
+log "Retrieving MAC address for interface $UAP_IF..."
+MAC=$(cat /sys/class/net/${UAP_IF}/address 2>> "$LOGFILE")
+log "MAC address retrieved: $MAC"
+
+# Reload WLAN modules
+log "Reloading WLAN modules..."
+modprobe -r dhd >> "$LOGFILE" 2>&1 && log "WLAN module 'dhd' removed successfully." || log "Failed to remove WLAN module 'dhd', continuing..."
+modprobe dhd >> "$LOGFILE" 2>&1 && log "WLAN module 'dhd' loaded successfully." || log "Failed to load WLAN module 'dhd'."
+sleep 2
```

Comment:
We should replace `sleep X` calls with `popup_timeout "(1/3) Loading WiFi module.." X` or similar, and reflect the success/failure of the current step.

Possibly use `X-1` instead of `X` as popup_timeout needs ~0.7s to load on NX500...